core/src/stages/generate_grasp_pose.cpp

```diff
@@ -78,13 +78,41 @@ void GenerateGraspPose::init(const core::RobotModelConstPtr& robot_model) {
 	const std::string& eef = props.get<std::string>("eef");
 	if (!robot_model->hasEndEffector(eef))
 		errors.push_back(*this, "unknown end effector: " + eef);
-	else {
-		// check availability of eef pose
-		const moveit::core::JointModelGroup* jmg = robot_model->getEndEffector(eef);
-		const std::string& name = props.get<std::string>("pregrasp");
-		std::map<std::string, double> m;
-		if (!jmg->getVariableDefaultPositions(name, m))
-			errors.push_back(*this, "unknown end effector pose: " + name);
+	else if (props.property("pregrasp").defined()) {
+		// if pregrasp pose is defined, check if it's valid
+		const moveit::core::JointModelGroup* eef_jmg = robot_model->getEndEffector(eef);
+		const boost::any& pregrasp_prop = props.get("pregrasp");
+		if (pregrasp_prop.type() == typeid(std::string)) {
+			// check if the specified pregrasp pose is a valid named target
+			const auto& pregrasp_name = boost::any_cast<std::string>(pregrasp_prop);
+			std::map<std::string, double> m;
+			if (!eef_jmg->getVariableDefaultPositions(pregrasp_name, m))
+				errors.push_back(*this, "pregrasp is set to unknown end effector pose: " + pregrasp_name);
+		} else if (pregrasp_prop.type() == typeid(sensor_msgs::JointState)) {
+			// check if the specified pregrasp pose is a valid named target
+			const auto& pregrasp_state = boost::any_cast<sensor_msgs::JointState>(pregrasp_prop);
+			if (pregrasp_state.name.size() == pregrasp_state.position.size() &&
+			    pregrasp_state.name.size() == pregrasp_state.velocity.size() &&
+			    pregrasp_state.name.size() == pregrasp_state.effort.size()) {
```

Comment:
Why would you require velocity/effort to be set?
Positions suffice, though dynamics *might* be specified to define the dynamic grasp behavior.

Reply:
Currently, the code only uses positions. Dynamic grasp behaviour comes into play if a trajectory would be specified. That's not the case.